### PR TITLE
ARIBA 2.14.6 Update and Adding `test` Stage

### PR DESCRIPTION
**Pull request by HarryHung**
- Update ARIBA to 2.14.6 
- Update included SPAdes to 3.15.5
- Include running built-in test as `test` Stage 

Also, tested the built test image with some samples that I have.  

-----

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by erinyoung** [comment]
I was going through the test, and is the version of spades alright to use with this version of ARIBA?

```
#10 [test 1/3] RUN ariba version
#10 1.418 ARIBA version: 2.14.6
#10 1.418 
#10 1.418 External dependencies:
#10 1.418 bowtie2	2.3.4.1	/usr/bin/bowtie2
#10 1.418 cdhit	4.7	/usr/bin/cd-hit-est
#10 1.418 nucmer	3.1	/usr/bin/nucmer
#10 1.418 spades	ERROR	/SPAdes-3.15.5-Linux/bin/spades.py
#10 1.418 
#10 1.418 External dependencies OK: True
#10 1.418 
#10 1.418 Python version:
#10 1.418 3.6.9 (default, Mar 10 2023, 16:46:00) 
#10 1.418 [GCC 8.4.0]
#10 1.418 
#10 1.418 Python packages:
#10 1.418 ariba	2.14.6	/usr/local/lib/python3.6/dist-packages/ariba/__init__.py
#10 1.418 bs4	4.12.2	/usr/local/lib/python3.6/dist-packages/bs4/__init__.py
#10 1.418 dendropy	4.6.1	/usr/local/lib/python3.6/dist-packages/dendropy/__init__.py
#10 1.418 pyfastaq	3.17.0	/usr/local/lib/python3.6/dist-packages/pyfastaq/__init__.py
#10 1.418 pymummer	0.10.3	/usr/local/lib/python3.6/dist-packages/pymummer/__init__.py
#10 1.418 pysam	0.15.4	/usr/local/lib/python3.6/dist-packages/pysam/__init__.py
#10 1.418 
#10 1.418 Python packages OK: True
#10 1.418 
#10 1.418 Everything looks OK: True
#10 1.418 WARNING: I tried to get the version of spades with: "/SPAdes-3.15.5-Linux/bin/spades.py --version" and the output didn't match this regular expression: "SPAdes\s+v([0-9\.]+)"
#10 DONE 1.6s
```

**Comment by HarryHung** [comment]
ARIBA documentation does not specify the required SPAdes version. I have tried a test run with `--assembler spades` without any problem though.

File: ariba/2.14.6/README.md
Comment (erinyoung) [inline ariba/2.14.6/README.md:6]: Can you add a section that lists the additional tools that were installed and their versions? I think it's just spades and pysam. An example README for comparison can be found at https://github.com/StaPH-B/docker-builds/tree/master/dnaapler/0.1.0

File: ariba/2.14.6/README.md
Comment (erinyoung) [inline ariba/2.14.6/README.md:37]: Could you include an example usage in your Example usage statement? This is intended to be an example command that someone would use inside a container.

File: ariba/2.14.6/Dockerfile
Comment (erinyoung) [inline ariba/2.14.6/Dockerfile:61]: Could you add some ARG values at the beginning the file and replace these with those values?

For example, only lines ~2 & 3
ARG PYSAM_VER="0.15.4"
ARG ARIBA_VER="2.14.6"

And then change this line to 
```
RUN pip3 install pysam==${PYSAM_VER} ariba==${ARIBA_VER}
```

File: ariba/2.14.6/Dockerfile
Comment (erinyoung) [inline ariba/2.14.6/Dockerfile:1]: Is there any reason to continue to use bionic? (There might be dependency issues that I'm not aware of by using jammy, so this is something I'm curious about and I likely won't make you change it)

File: ariba/2.14.6/Dockerfile
Comment (erinyoung) [inline ariba/2.14.6/Dockerfile:16]: Can this ENV be changed to an ARG? ENV persist, and we try not to include any of those if we don't have to.

**Comment by erinyoung** [comment]
As long as it works with the version of spades that was installed, I think it's fine. I just wanted you to be aware of the warning that gets printed.

**Comment by erinyoung** [comment]
@kapsakcj , I don't use ARIBA very often. Are there any additional concerns that should be brought up for this Dockerfile for version 2.14.6?

File: ariba/2.14.6/Dockerfile
Comment (HarryHung) [reply ariba/2.14.6/Dockerfile]: Done

File: ariba/2.14.6/Dockerfile
Comment (HarryHung) [reply ariba/2.14.6/Dockerfile:1]: The original Dockerfile for `2.14.4` by @kapsakcj  used `ubuntu:bionic` and worked fine. Just trying to make minimum changes to update the image to `2.14.6`. Any benefit updating to Jammy?

**Comment by HarryHung** [comment]
Made changes based on your comments.

**Comment by erinyoung** [comment]
I think that everything looks good. 

I did make some adjustments:
1. converted the spades version to an ARG value
2. added a link in your readme to the files (this currently isn't in the documentation for this repo)
3. added a 'CMD' with the ariba help function (this also is not currently in the documentation for this repo. What it does is give a default command for anyone that runs containers from the built image.)

I would like to wait until I get @kapsakcj opinion as I do not use ARIBA personally and he is listed as a maintainer in the Dockerfile.

**Comment by erinyoung** [comment]
This PR should close two issues : https://github.com/StaPH-B/docker-builds/issues/392 and https://github.com/StaPH-B/docker-builds/issues/518

File: ariba/2.14.6/Dockerfile
Comment (kapsakcj) [reply ariba/2.14.6/Dockerfile:1]: We generally prefer to use more recent Ubuntu releases, e.g. jammy and focal, but bionic should continue to receive extended support & security updates through 2028 https://ubuntu.com/about/release-cycle

It's probably not worth the hassle to upgrade to jammy or focal as it may break some of the dependencies installed via `apt-get` since those packages names can change slightly between ubuntu releases.

I'm fine with leaving as ubuntu:bionic, but if a simple upgrade to ubuntu:jammy is possible without too much frustration, I would say let's upgrade.

**Comment by kapsakcj** [comment]
This PR looks great, thank you @HarryHung . I see that the tests pass which is a welcome addition to the dockerfile.

The only other thing that was on my mind is that the dockerfile does not download and prepare and reference files/databases. 

Maybe I'm not the best person to ask because I don't use `ariba` much at all, but I thought it would be convenient to include some of the databases along with the software, pending that they aren't too big in filesize.

I attempted this when I first tried writing this dockerfile, but ran into issues/errors with specific databases. Some worked, some did not. See my code & comments here: https://github.com/StaPH-B/docker-builds/blob/b27bee7f30b5c3f90cd838cd15a5795a86204da9/ariba/2.14.6/Dockerfile#L77

If it's too much of a hassle to add this to the dockerfile/docker image, that's fine and we can leave that part out. It will just be up to the user to download and prepare their reference files

**Comment by HarryHung** [comment]
@kapsakcj Sadly, I provide my own custom FASTA + metadata when I use ARIBA instead of using one of the databases, therefore I am not familiar with that part.

**Comment by kapsakcj** [comment]
No worries, I'm fine to merge the PR as is. No need to include databases. If a user requests this in the future, we can make a new docker image that includes it.

I attempted to upgrade the base to ubuntu:jammy but ran into some issues with installing pysam via pip, so I'm not going to bother troubleshooting further. We can stick with ubuntu:bionic

**Review by kapsakcj — APPROVED** [review]

**Comment by kapsakcj** [comment]
Once the test passes, I'll merge and deploy the docker image to dockerhub and quay. The image name & tag will be `staphb/ariba:2.14.6` and we will overwrite `staphb/ariba:latest` tag

**Comment by kapsakcj** [comment]
Thanks for the PR, Harry!

You can follow the deploy workflow here: https://github.com/StaPH-B/docker-builds/actions/runs/5623857235

Should be available on dockerhub and quay in the next 15 min or so

**Comment by HarryHung** [comment]
Thank you for checking and deploying.

**Pull request merged**
